src/rules/classNameRule.ts

```diff
@@ -36,38 +37,23 @@ export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "Class name must be in pascal case";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new NameWalker(sourceFile, this.getOptions()));
+        return this.applyWithFunction(sourceFile, walk);
     }
 }
 
-class NameWalker extends Lint.RuleWalker {
-    public visitClassDeclaration(node: ts.ClassDeclaration) {
-        // classes declared as default exports will be unnamed
-        if (node.name != null) {
-            const className = node.name.getText();
-            if (!this.isPascalCased(className)) {
-                this.addFailureAtNode(node.name, Rule.FAILURE_STRING);
+function walk(ctx: Lint.WalkContext<void>) {
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+        if (isClassLikeDeclaration(node) && node.name !== undefined ||
+            isInterfaceDeclaration(node)) {
+            if (!isPascalCased(node.name!.text)) {
```

Comment:
ehh, i'm not too worried about this. usually when this happens it's done to help promote readability of the conditions. large conditional expressions are scary